digest/reader.go

```diff
@@ -180,7 +181,7 @@ type readerWithDigest struct {
 func NewReaderWithDigest(reader io.Reader) ReaderWithDigest {
 	return &readerWithDigest{
 		reader: reader,
-		digest: NewDigest(),
+		digest: adler32.New(),
```

Comment:
Same here.  Since these aren't highly concurrent and we cache the use of them (in the readers, seekers, etc), we don't have to use the stack based adler32 approach - we can use the stdlib which is a few percent faster than the stack based adler32 approach (although stack based is zero-alloc, it'll only be fast in the highly concurrent use case).